### DISCUSSION TITLE
Training on multiple concepts

### DISCUSSION THREAD

**Discussion by muhammad-ahmed-ghani** [TheLastBen/fast-stable-diffusion#1299] [Q&A]
Hi @TheLastBen , I have a question related to optimal settings for training multiple concept means (on couple). I tried with 20 male and 20 female images and trained it on (text-2800, unet-4000) steps. I got bad results. Learning rate was 2e-6. Can you please help me with training on couple images ?
I also test with 10+10 images but results are pretty much the same.

Model: SD v1.5

Tried Settings:
- text-350, unet-4000 (learning rate 5e-6)
- text-900, unet-4000 (learning rate 2e-6)
- text-1400, unet-2000 (learning rate 2e-6)
- text-2800, unet-4000 (learning rate 2e-6)

**Comment by cdefghijkl** [comment]
What exactly the bad results is it? Not looking same as the instance images or kinda similar but skewed/distorted?

**Reply by muhammad-ahmed-ghani** [reply]
> What exactly the bad results is it? Not looking same as the instance images or kinda similar but skewed/distorted?

The faces are mostly distorted and out of 10 only 1 image has 50% similarity of the couple. I trained the model on single face and it works well but on multiple concepts I am having the issue. Using **DDIMScheduler** in the model.

**Reply by cdefghijkl** [reply]
I think it's undertraining, Try change both the unet and text encoder's batch size to 8 and lr scheduler to constant, then unet step 500, text encoder 50. If the generated image still not looking similar, resume training only unet 100 step, repeat until you get what you want.

**Reply by cdefghijkl** [reply]
use 5e-6 for unet and 5e-7 for text encoder.

**Reply by muhammad-ahmed-ghani** [reply]
> > What exactly the bad results is it? Not looking same as the instance images or kinda similar but skewed/distorted?
> 
> The faces are mostly distorted and out of 10 only 1 image has 50% similarity of the couple. I trained the model on single face and it works well but on multiple concepts I am having the issue. Using **DDIMScheduler** in the model.

I tried with your settings but results are much worse now.

With text=2800, unet=4000, learning-rate=2e-6
![198_512x512](https://user-images.githubusercontent.com/63394104/211297429-cc50d93f-eae9-4020-aa5e-fd3fc2599660.jpg)


With text=50, unet=600, learning-rate= unet-5e-6, text-5e-7, batch-size=8
![7_512x512](https://user-images.githubusercontent.com/63394104/211297642-42163fbd-1b8c-4a5c-9c9a-519b4bad62eb.jpg)

**Reply by cdefghijkl** [reply]
It works for me, idk why. Glad you can did it better.

**Reply by cdefghijkl** [reply]
Could you try my other setting?
but this setting is for 40 images for one person: unet 300 steps LR 2e-5, text encoder 35 steps LR 1e-6, batch size 10 for both (so it will be like 75 unet step batch 1 for one image).
Use Hassanblend 1.4 or openjourney-v2-unpruned as your base model.

**Reply by TheLastBen** [reply]
40 images for one person is too much, keep it below 15 and the unet lr below 1e-5 for the v1.5 and 2e-5 for the v2

**Reply by cdefghijkl** [reply]
I know its too much but it works for me, I will try train with lesser image later.
The only thing I have a hard time to apply is how to not overfit and can apply many art style to the trained person, when I prompt "Painting of [filename] blablabla..." Its not showing painting but still a photo realistic.

**Reply by TheLastBen** [reply]
to solve that, limit the text encoder steps to maximum 400 steps, the unet steps won't affect styling

**Reply by cdefghijkl** [reply]
400 for one person?

**Reply by TheLastBen** [reply]
text encoder : 400 steps, 1e-6 for 15 images of one subject.
